devops/scripts/deploy-decoupled-upstream.sh

```diff
@@ -57,11 +60,13 @@ for commit in $newcommits; do
     echo "You may wish to ensure that nothing in this commit is meant for release."
     delete=(${commit})
     for remove in "${delete[@]}"; do
-      for i in "${commits[@]}"; do
-        if [ [ ${commits[i]} = $remove ]]; then
-          unset 'commits[i]'
-        fi
-      done
+      if (( ${#commits[@]} )); then
+        for i in "${commits[@]}"; do
+          if [[ ${commits[0]} = $remove ]]; then
```

Comment:
I believe this should also be `==` but it was single before, and per your above comments now I'm not 100% sure— the `[ [` vs `[[` may also have significance, I'm not confident enough in my bash syntax to be sure what is or isn't correct—

If I was writing this new I would do
```suggestion
          if [[ "${commits[0]}" == "$remove" ]]; then
```
but that doesn't mean the current code is wrong, either.

Reply:
The `[[` vs `[ [` did have practical changes in behavior although I couldn't explain why exactly. (tbh, after reading up on some of this stuff, I'm not convinced brackets are even required for conditionals in bash...)

I think after some amount of research this morning `=` vs `==` does not make a difference if there's spaces between the variable and the operator and the assignment.